modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/SelectionFilter.js

```diff
@@ -44,6 +45,21 @@ function fetchData(apiURL, searchParam, currentPage = 1) {
 	}).then((response) => response.json());
 }
 
+const mapResponseData = (data, itemLabelProp, itemKey) => {
+	return {
+		...data,
+		items: data.items.map((item) => {
+			const option = {label: undefined, value: undefined};
+			option.label = itemLabelProp
+				? getValueFromItem(item, itemLabelProp)
+				: item.label;
+			option.value = itemKey ? item[itemKey] : item.value;
+
+			return option;
+		}),
+	};
+};
+
```

Comment:
Several things here:

1. It is typically a good practice to have an object as a single argument. That makes maintenance easier:
    - it's easier to add or remove arguments
    - you can omit arguments and avoid awkward nulls
    - it forces devs to name things consistently, as object properties need to be the same 
    - you can deconstruct args, if you like
1. For clarification, we can split off `response` from items array. Instead of `...data`, we can directly have `setTotal(response.totalCount);`
1. `mapResponseData` is a bit off, but you can avoid naming woes by placing map inline in `loadSelectionItems`. If you still want it, maybe because of too many line breaks, it should probably be something like `getSelectionItems`.
1. You can simplify internal logic, to something like 
```js
const selectionItems = response.items.map((item) => {
	return {
		label: itemLabelProp
			? getValueFromItem(item, itemLabelProp)
			: item.label,
		value: itemKey ? item[itemKey] : item.value,
	};
});
```